spec/acceptance/resolved_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'systemd' do
+  context 'configure systemd resolved' do
+    it 'works idempotently with no errors' do
+      pp = <<-PUPPET
+      class{'systemd':
+        manage_resolved    => true,
+        manage_resolv_conf => #{default[:hypervisor] != 'docker'},
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 and newer installs package first run before fact  $facts['internal_services'] is set
+      apply_manifest(pp, catch_failures: true) if !%w[7 8].include?(fact('os.release.major')) && (fact('os.family') == 'RedHat')
+      apply_manifest(pp, catch_changes: true)
+    end
+
+    # RedHat 7 does not have systemd-resolved available at all.
```

Comment:
On CentOS 7 there is a `systemd-resolved` package with a `/usr/lib/systemd/system/systemd-resolved.service`. In practice I noticed it was harder to use due to a lack of `resolvectl` so I ended up not using it, but it's not entirely correct. I'd be surprised if RHEL 7 and CentOS 7 differed in this.

Reply:
Indeed that package  is available CentOS 7 also, just not in 8. :-(

Reply:
Does that mean we should include the package name in the Hiera data?